mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/ElasticSearchMailboxIndexerTest.java

```diff
@@ -39,7 +48,7 @@
 
 import com.google.common.collect.Lists;
 
-public class ElasticSearchIndexerTest {
+public class ElasticSearchMailboxIndexerTest {
```

Comment:
Why isn't this test generic?

I mean we are not indexing emails here, right? What you are really testing is the expected behaviour of ElasticSearchIndexer.

Reply:
It's a bit of over engineering, IMO.

We should do that when needed, not before.

Reply:
Well, I disagree.

 - 1. It's needed in that Sprint
 - 2. That was a generic component before it was changed
 - 3. We do care about only customizing the Index / mapping used but honestly this class is orthogonal to the document structure.

What about keeiping the previous ElasticSearchIndexer as is (without injection) and have several `Supplier<ElasticSearchIndexer>` giving us the right indexers, configured with the right index/mapping, for instance for the `mailbox` context?